unittests/test-contracts/savanna/common/bitset.hpp

```diff
@@ -0,0 +1,82 @@
+using namespace eosio;
+
+namespace savanna {
+
+    class bitset {
+        using word_t = uint8_t;
+        static constexpr size_t num_bits = sizeof(word_t) * 8;
+        static size_t round_up(size_t sz) { return (sz + num_bits - 1) / num_bits; }
+
+    public:
+
+        // compare two bitsets
+        static std::pair<savanna::bitset, savanna::bitset> compare(const savanna::bitset& bs1, const savanna::bitset& bs2) {
+            check(bs1.size() == bs2.size(), "bitsets must be of the same size");
+
+            size_t size = bs1.size();
+            std::vector<word_t> intersection_data(bs1.data.size());
+            std::vector<word_t> symmetric_difference_data(bs1.data.size());
+
+            for (size_t i = 0; i < bs1.data.size(); ++i) {
+                intersection_data[i] = bs1.data[i] & bs2.data[i];
+                symmetric_difference_data[i] = bs1.data[i] ^ bs2.data[i];
+            }
+
+            savanna::bitset intersection(size, intersection_data);
+            savanna::bitset symmetric_difference(size, symmetric_difference_data);
+
+            return {intersection, symmetric_difference};
+        }
+
+        bitset(word_t size) 
+            : data(round_up(size), 0) {}
+        
+        bitset(word_t size, const std::vector<word_t> raw_bitset)
+           : data(raw_bitset) {
+               check(raw_bitset.size() == round_up(size), "invalid raw bitset size");
+           }
+
+        // Set a bit to 1
+        void set(word_t index) {
+            check_bounds(index);
+            data[index / 32] |= (1UL << (index % num_bits));
+        }
+
+        // Clear a bit (set to 0)
+        void clear(word_t index) {
+            check_bounds(index);
+            data[index / num_bits] &= ~(1UL << (index % num_bits));
+        }
+
+        // Check if a bit is set
+        bool test(word_t index) const {
+            check_bounds(index);
+            return (data[index / num_bits] & (1UL << (index % num_bits))) != 0;
+        }
+
+        // Size of the bitset
+        word_t size() const {
+            return num_bits;
+        }
+        
+        std::string to_string() const {
+            const char* hex_chars = "0123456789abcdef";
+            std::string result;
+            result.reserve(data.size() * 2); // Each byte will be represented by two hex characters
+            for (const auto& byte : data) {
+                result.push_back(hex_chars[(byte >> 4) & 0x0F]);
+                result.push_back(hex_chars[byte & 0x0F]);

```

Comment:
I think these two lines should be swapped
```suggestion
                result.push_back(hex_chars[byte & 0x0F]);
                result.push_back(hex_chars[(byte >> 4) & 0x0F]);
```